libtiledbsoma/src/utils/arrow_adapter.h

```diff
@@ -702,8 +702,61 @@ class ArrowAdapter {
         return arrow_array;
     }
 
-    // These table-column getters are, as of this writing, intended primarily
-    // for keystroke-reduction in unit-test cases.
+    // Given TileDB 8-bit booleans, packs them to 1-bit Arrow-style booleans.
+    static ArrowArray* make_arrow_array_child_bool(
+        const std::vector<uint8_t>& v) {
+        // Use malloc here, not new, to match ArrowAdapter::release_array
+        auto arrow_array = (ArrowArray*)malloc(sizeof(ArrowArray));
```

Comment:
possibly naive question:  if we are using nanoarrow, why not use the nanoarrow array allocation routines?

(see: https://arrow.apache.org/nanoarrow/0.2.0/c.html)

Reply:
Apparently we have not had the pattern of using that. TIL, thank you for the feedback.

If you feel it would be worth the dev time to refactor our code to use that API in this library, please file a tracking issue.

Thanks again @bkmartinjr !

Reply:
I'm not going to hold up this PR.  But this is exactly what got us into the Arrow offset bug-fest previously. If we would use the provided API, I believe these issues would be less likely to occur.

So yes, we should be using a framework (I thought we were!).  I see this as your decision as eng lead, so I just leave you with this thought.